Library/Homebrew/rubocops/conflicts_cop.rb

```diff
@@ -0,0 +1,33 @@
+require_relative "./extend/formula_cop"
+require_relative "../extend/string"
+
+module RuboCop
+  module Cop
+    module FormulaAudit
+      # This cop audits versioned Formulae for `conflicts_with`
+      class Conflicts < FormulaCop
+        MSG = <<-EOS.undent
+            Versioned formulae should not use `conflicts_with`.
+            Use `keg_only :versioned_formula` instead.
+        EOS
+
+        def audit_formula(_node, _class_node, _parent_class_node, body)
+          return unless versioned_formula?
+          whitelist = %w[node@ bash-completion@].freeze
```

Comment:
Let's make this a constant earlier in the class and have one formula per line so they stand out a bit more